### DISCUSSION TITLE
Use Buffers in the MxAPI Path Functions

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#3902] [Development]
The Mailbox API has four path functions.
Converting these to use `struct Buffer *` will help the general move to use Buffers for paths.

```c
struct MxOps
{
  int path_canon   (char *buf, size_t buflen);
  int path_pretty  (char *buf, size_t buflen, const char *folder);
  int path_parent  (char *buf, size_t buflen);
  int path_is_empty(const char *path);
};
```

Each of these four functions is implemented by each mailbox backend, e.g.
- Imap: `imap_path_canon()`, `imap_path_pretty()`, etc
- Maildir: `maildir_path_canon()`, `maildir_path_pretty()`, etc
- etc

These are called, generally, through wrappers in `mx.c`

**Comment by roccoblues** [comment]
Forgot to mention this discussion in the PR. `path_canon()` and `path_pretty()` have been done in here: https://github.com/neomutt/neomutt/pull/3906

Working on `path_is_empty()` now.

**Comment by roccoblues** [comment]
@flatcap I guess this is finished and can be closed too?
